commit d77f5ee722596a818b4e92f536e5a3a20afac528 (from ecf3ab5de457df5d73141ecd8a98cc528cccc3b4)
Merge: ecf3ab5 fde17d5
Author: daid <daid303@gmail.com>
Date:   Fri Jun 28 08:04:55 2013 -0700

    Merge pull request #532 from codexmas/ServoEndstops
    
    Fix for ugly code compilation errors in SERVO_ENDSTOP checks

diff --git a/Marlin/Marlin_main.cpp b/Marlin/Marlin_main.cpp
index b552d20..0677c0b 100644
--- a/Marlin/Marlin_main.cpp
+++ b/Marlin/Marlin_main.cpp
@@ -687,10 +687,12 @@ static void homeaxis(int axis) {
       0) {
 
     // Engage Servo endstop if enabled
-    #ifdef SERVO_ENDSTOPS[axis] > -1
-      servos[servo_endstops[axis]].write(servo_endstop_angles[axis * 2]);
+    #ifdef SERVO_ENDSTOPS
+      if (SERVO_ENDSTOPS[axis] > -1) {
+        servos[servo_endstops[axis]].write(servo_endstop_angles[axis * 2]);
+      }
     #endif
-
+      
     current_position[axis] = 0;
     plan_set_position(current_position[X_AXIS], current_position[Y_AXIS], current_position[Z_AXIS], current_position[E_AXIS]);
     destination[axis] = 1.5 * max_length(axis) * home_dir(axis);
@@ -715,8 +717,10 @@ static void homeaxis(int axis) {
     endstops_hit_on_purpose();
 
     // Retract Servo endstop if enabled
-    #ifdef SERVO_ENDSTOPS[axis] > -1
-      servos[servo_endstops[axis]].write(servo_endstop_angles[axis * 2 + 1]);
+    #ifdef SERVO_ENDSTOPS
+      if (SERVO_ENDSTOPS[axis] > -1) {
+        servos[servo_endstops[axis]].write(servo_endstop_angles[axis * 2 + 1]);
+      }
     #endif
   }
 }

